test/extended/authorization/per_node_validatingadmissionpolicy.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "only-allow-name-matching-node-configmaps"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   [""]
+        apiVersions: ["v1"]
+        # notice you cannot restrict DELETEs.  Your per-node actor should not have deletion powers.
+        # it probably shouldn't have create powers either, but we can control that
+        operations:  ["CREATE", "UPDATE"]
+        resources:   ["configmaps"]
+  variables:
+    - name: hasNodeName
+      expression: ('authentication.kubernetes.io/node-name' in request.userInfo.extra)
+    - name: isPartitionedServiceAccount
+      expression: request.userInfo.username == "system:serviceaccount:e2e-ns:default"
+  validations:
+    - expression: "!variables.isPartitionedServiceAccount || (!variables.hasNodeName) || object.metadata.name == request.userInfo.extra[\"authentication.kubernetes.io/node-name\"][0]"
```

Comment:
object.metadata.name could be any field you want and both the oldObject and the newObject.

Reply:
Shouldn't the last check be  `!=` ?

Reply:
> Shouldn't the last check be `!=` ?

When the expression results in true, the request is allowed.  If the metadata.name == the node name of the allowed pod SA, then we want to allow the request.  If it doesn't match, then fail.

I think both sides of this are tested in the e2e test itself.